### PR TITLE
Compatibility updates for v2 in README

### PR DESCRIPTION
**Pull request by jkowall**
## Which problem is this PR solving?
Addresses part of https://github.com/jaegertracing/documentation/issues/117

## Description of the changes
Update text to match Jaeger v2

## How was this change tested?
Not tested

## Checklist
- [X] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [X] I have signed all commits

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ For example, any given Jaeger installation at Uber is typically processing sever
 
 ### Relationship with OpenTelemetry
```
Comment (yurishkuro) [inline README.md:68]: Hm, I recently removed this section altogether, moved it to the documentation.

https://www.jaegertracing.io/docs/next-release-v2/#relationship-with-opentelemetry

File: README.md [hunk 2]
```diff
@@ -124,19 +124,19 @@ of routing the traffic from Zipkin libraries to the Jaeger backend.
 
 ## Version Compatibility Guarantees
 
-Occasionally, CLI flags can be deprecated due to, for example, usability improvements or new functionality.
+Since Jaeger uses many components from the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector/) we try to maintain configuration compatibility between Jaeger releases. Occasionally, configuration elements in Jaeger (or in Jaeger v1 CLI flags) can be deprecated due to usability improvements, new functionality, or changes in our dependencies.
```
Comment (yurishkuro) [inline README.md:127]: I think "configuration option" used in L129 sounds better than "element", perhaps converge to that one for consistency?

**Review by yurishkuro — COMMENTED** [review]
changes lgtm

**Comment by yurishkuro** [comment]
@jkowall can you update per comments?

File: README.md (hunk 2 above)
Comment (jkowall) [reply README.md:127]: Should be resolved in latest push

**Comment by yurishkuro** [comment]
Ωconflicts w/ main

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**